plugins/chromatic_correction.py

```diff
@@ -0,0 +1,252 @@
+        """
+        Compute affine transformation between a reference channel and another coloured channel.
+        :param ref_channel: (model.DataArray) The reference channels to which other channels are aligned
+        :param channels: (list of model.DataArray) The channels are transformed to align with reference channel
+        :param max_distance: (float in meters) The maximum distance accepted to match two same locations in between
+         reference channel and channels
+        """
+        trans_per_channel = {model.MD_CHROMATIC_COR: {}}  # stores transformation values per channel
+        dist_pxl = int(max_distance / ref_channel.metadata[model.MD_PIXEL_SIZE][0])  # convert m to pixel units
+        centers_ref = MaximaFind(ref_channel, BEAD_QUANTITY, dist_pxl)
+
+        for i, im in enumerate(channels):
+            # Find corresponding points
+            centers_new = MaximaFind(im, BEAD_QUANTITY, dist_pxl)
+            corresponding_pairs = find_corresponding_points(centers_ref, centers_new, dist_pxl)
+
+            # Extract the matching points
+            points1 = centers_ref[corresponding_pairs[:, 0]]
+            points2 = centers_new[corresponding_pairs[:, 1]]
+
+            if len(corresponding_pairs) < 4:
+                txt = ('The chromatic correction did not happen for channel with %s. Skipping this channel.\n\n'
+                       'Alternatively, navigate to location with more beads or change the bead diameter in the script'
+                       % im.metadata[model.MD_DESCRIPTION])
+                box = wx.MessageDialog(self.main_app.main_frame,
+                                       txt, "Partial failure of chromatic correction", wx.OK)
+                box.ShowModal()
+                box.Destroy()
+
+            else:
+                # Estimate the transformation matrix
+                affine = AffineTransform.from_pointset(points1, points2)
+                mat = numpy.eye(3)
+                mat[:2, :2] = affine.matrix
+                mat[:2, 2] = affine.translation
+                ref_mat = numpy.eye(3)
+                mat = ref_mat @ mat
+
+                # Scale is the norm of the first two columns
+                sx = numpy.linalg.norm(mat[:2, 0])
+                sy = numpy.linalg.norm(mat[:2, 1])
+                scale2 = numpy.array([sx, sy])
+
+                # Compute rotation
+                rotation = numpy.arctan2(mat[1, 0] / sx, mat[0, 0] / sx)
+
+                # compute scale
+                scale = (1 / scale2[0], 1 / scale2[1])
+
+                # compute shear
+                a, b = mat[0, 0], mat[0, 1]
+                c, d = mat[1, 0], mat[1, 1]
+                shear = numpy.arctan2(a * c + b * d, scale[0] * scale[1])
+
+                # compute translation
+                tx = mat[0, 2]
+                ty = mat[1, 2]
+                pixel_size = im.metadata[model.MD_PIXEL_SIZE]
+                translation = (tx * pixel_size[0], -ty * pixel_size[1])
```

Comment:
Please move this to a separate function which takes points1 & points2, and returns scale, translation, rotation, and shear.